pennylane/ops/channel.py

```diff
@@ -377,6 +378,104 @@ def _kraus_matrices(cls, *params):
 
 
+class PauliError(Channel):
+    r"""PauliError(operators, p, wires)
+    Arbitrary number qubit, arbitrary Pauli operator error channel.
+
+    This channel is modelled by the following Kraus matrices:
+
+    .. math::
+        K_0 = \sqrt{1-p} * I
+
+    .. math::
+        K_1 = \sqrt{p} * (K_{w0} \otimes K_{w1} \otimes \dots K_{wn})
+
+    Where :math:`I` is the Identity,
+    and :math:`\otimes` denotes the Kronecker Product,
+    and :math:`K_{wi}` denotes the Kraus matrix corresponding to the operator acting on wire :math:`wi`,
+    and :math:`p` denotes the probability with which the channel is applied.
+
+    .. warning::
+
+        The size of the Kraus matrices for PauliError scale exponentially
+        with the number of wires, the channel acts on. Simulations with
+        PauliError can result in a significant increase in memory and
+        computational usage. Use with caution!
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 3
+
+    Args:
+        operators (str): The Pauli operators acting on the specified (groups of) wires
+        p (float): The probability of the operator being applied
+        wires (Sequence[int] or int): The wires the channel acts on
+
+    **Example:**
+
+    >>> pe = PauliError("X", 0.5, wires=0)
+    >>> km = pe.kraus_matrices
+    >>> km[0]
+    array([[0.70710678, 0.        ],
+           [0.        , 0.70710678]])
+    >>> km[1]
+        array([[0.        , 0.70710678],
+               [0.70710678, 0.        ]])
+    """
+
+    num_params = 2
+    num_wires = AnyWires
+    par_domain = "L"
+
+    ops = {
+        "X": np.array([[0, 1], [1, 0]]),
+        "Y": np.array([[0, -1j], [1j, 0]]),
+        "Z": np.array([[1, 0], [0, -1]]),
+    }
+
+    def __init__(self, *params, wires=None, do_queue=True):
+        super().__init__(*params, wires=wires, do_queue=do_queue)
+        operators, p = params[0], params[1]
+
+        # check if the specified operators are legal
+        if not all(c in "XYZ" for c in operators):
+            raise ValueError("The specified operators need to be either of 'X', 'Y' or 'Z'")
+
+        # check if probabilities are legal
+        if not 0.0 <= p <= 1.0:
+            raise ValueError("p must be between [0,1]")
+
+        # check if the number of operators matches the number of wires
+        if (type(wires) == int and len(operators) != 1) or (
+            type(wires) == list and len(operators) != len(wires)
+        ):
+            raise ValueError("The number of operators must match the number of wires")
+
+        nq = len(operators)
+
+        if nq > 20:
+            warnings.warn(
+                f"The resulting Kronecker matrix will have dimensions {2**(nq)} x {2**(nq)}.\nThis equals {2**nq*2**nq*8/1024**3} GB of physical memory."
+            )
+
+    @classmethod
+    def _kraus_matrices(cls, *params):
+        operators, p = params[0], params[1]
+
+        nq = len(operators)
```

Comment:
```suggestion
        nq = len(self.wires)
```

Reply:
This doesn't work

Reply:
I think, that's the same problem we had in the beginning

Reply:
Oh, good catch! We don't have `self` here indeed.